controllers/useraccount/console_usersettings.go

```diff
@@ -0,0 +1,204 @@
+package useraccount
+
+import (
+	"context"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	ConsoleUserSettingsIdentifier         = "console.openshift.io/user-settings"
+	ConsoleUserSettingsUID                = "console.openshift.io/user-settings-uid"
+	UserSettingNS                         = "openshift-console-user-settings"
+	ConsoleUserSettingsResourceNamePrefix = "user-settings-"
+)
+
+// deleteConfigMap deletes a ConfigMap associated with a user from console setting.
+// It first attempts to find the ConfigMap by name. If not found, it then looks for the ConfigMap by label.
+// If multiple ConfigMaps are found by label, all of them are deleted.
+// The function returns an error if any occurred.
+func deleteConfigMap(ctx context.Context, cl client.Client, userUID string) error {
+	name := ConsoleUserSettingsResourceNamePrefix + userUID
+	logger := log.FromContext(ctx)
+	logger.Info(fmt.Sprintf("deleting configmap with name %s", name))
+	var toDelete []client.Object
+
+	// Create a ConfigMap object type to use with getConsoleUserSettingObjectByName function
+	objectType := &corev1.ConfigMap{TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"}}
+
+	// Attempt to find the ConfigMap by name
+	configMap, err := getConsoleUserSettingObjectByName(ctx, cl, name, objectType)
+	if err == nil && configMap == nil {
+		// could not find the CM by name, try finding the configmap by label
+		if configMapList, err := getConfigMapByLabel(ctx, cl, userUID); err != nil || len(configMapList) == 0 {
+			return err
+		} else if len(configMapList) > 0 {
+			// if the number of items retrieved are more than one, should delete all of them
+			for key := range configMapList {
+				toDelete = append(toDelete, &configMapList[key])
+			}
+		}
+	} else if configMap != nil {
+		toDelete = append(toDelete, configMap)
+	} else {
+		// there was some kind of error in getting
+		return err
+	}
```

Comment:
minor - another approach would be to check the error as first thing:
```
if err != nil {
    return err
}
// the other checks here ..

```

Reply:
thanks, simplified in [d4bc88c](https://github.com/codeready-toolchain/member-operator/pull/568/commits/d4bc88cc583bfad7f84185b5fa341ffc508984e3)